tests/unit/base/trained_model_serialization_unittest.cc.jinja2

```diff
@@ -111,64 +132,103 @@ TEST_F(TrainedModelSerializationTest, LinearMachine_{{class}})
 //	machine->set_features(NULL);
 //	machine->set_labels(NULL);
 
-	CLabels* predictions=machine->apply(test_feats);
+	auto predictions=Some<CLabels>(machine->apply(test_feats));
 
 	std::string filename;
 	ASSERT_TRUE(serialize_machine(machine, filename));
 
-	auto deserialized_machine=new {{class}}();
+	auto deserialized_machine=some<{{class}}>();
 	ASSERT_TRUE(deserialize_machine(deserialized_machine, filename));
 
-	CLabels* deserialized_predictions=deserialized_machine->apply(test_feats);
-
-	float64_t accuracy=1e-13;
+	auto deserialized_predictions=Some<CLabels>(deserialized_machine->apply(test_feats));
 	ASSERT(predictions->equals(deserialized_predictions, accuracy, true))
-
-	SG_FREE(machine);
-	SG_FREE(deserialized_machine);
-	SG_FREE(predictions);
-	SG_FREE(deserialized_predictions);
 }
 {%- endmacro %}
 
 {% macro kernel_machine_test(class) -%}
-TEST_F(TrainedModelSerializationTest, KernelMachine_{{class}})
+{% for store_model_features in ["true", "false"] -%}
+{% if store_model_features == "true" -%}
+{% set test_name = class + "_store_model_features" -%}
+{% else -%}
+{% set test_name = class-%}
+{% endif -%}
+TEST_F(TrainedModelSerializationTest, {{test_name}})
 {
-	auto machine=new {{class}}();
+	auto machine=some<{{class}}>();
 	load_data(machine->get_machine_problem_type());
 
-	CGaussianKernel *kernel = new CGaussianKernel(2.0);
+	CGaussianKernel *kernel=new CGaussianKernel(2.0);
 	kernel->init(train_feats, train_feats);
 	machine->set_kernel(kernel);
 	machine->set_labels(train_labels);
 
 	bool train_success=machine->train();
 	ASSERT_TRUE(train_success);
 
-	CLabels* predictions=machine->apply(test_feats);
+	auto predictions=Some<CLabels>(machine->apply(test_feats));
 
 	std::string filename;
-	ASSERT_TRUE(serialize_machine(machine, filename));
+	ASSERT_TRUE(serialize_machine(machine, filename, {{store_model_features}}));
 
-	auto deserialized_machine=new {{class}}();
+	auto deserialized_machine=some<{{class}}>();
 	ASSERT_TRUE(deserialize_machine(deserialized_machine, filename));
 
-	CLabels* deserialized_predictions=deserialized_machine->apply(test_feats);
-
-	float64_t accuracy=1e-13;
+	auto deserialized_predictions=Some<CLabels>(deserialized_machine->apply(test_feats));
 	ASSERT(predictions->equals(deserialized_predictions, accuracy, true))
+}
+{% endfor %}
+{%- endmacro %}
 
-	SG_FREE(machine);
-	SG_FREE(deserialized_machine);
-	SG_FREE(predictions);
-	SG_FREE(deserialized_predictions);
+{% macro gaussian_process_test(class) -%}
+TEST_F(TrainedModelSerializationTest, {{class}})
+{
+	auto machine=some<{{class}}>();
+	load_data(machine->get_machine_problem_type());
+
+	CGaussianKernel* kernel=new CGaussianKernel(10, 2);
+	CZeroMean* mean=new CZeroMean();
+
+	CLikelihoodModel* lik;
+	if (machine->get_machine_problem_type() == PT_REGRESSION)
+		lik=new CGaussianLikelihood(1.0);
+	else
+		lik=new CProbitLikelihood();
+
+	CSingleLaplaceInferenceMethod* inf=new CSingleLaplaceInferenceMethod(
```

Comment:
for classification you can use laplace, for regression maybe use exact
but the other approximations should also be tested, FITC, etc. See the notebooks